DonjinKrawler-Client/src/main/java/donjinkrawler/Client.java

```diff
@@ -43,58 +45,17 @@ private void setupKryo() throws IOException {
+        Client client = this;
 
+        kryoClient.addListener(new Listener() {
             public void received(Connection connection, Object object) {
-                if (object instanceof MessagePacket) {
-                    handleMessagePacket((MessagePacket) object);
-                } else if (object instanceof MapPacket) {
-                    MapPacket mapPacket = (MapPacket) object;
-                    if (mapPacket.update) {
-                        game.updateMap(mapPacket.rooms);
-                    } else {
-                        rooms = mapPacket.rooms;
-                    }
-                } else if (object instanceof IdPacket) {
-                    handleIdPacket((IdPacket) object);
-                } else if (object instanceof MoveCharacter && game != null) {
-                    MoveCharacter msg = (MoveCharacter) object;
-                    game.changeShellPosition(msg);
-                } else if (object instanceof EnemyPacket && game != null) {
-                    EnemyPacket enemyPacket = (EnemyPacket) object;
-                    if (enemyPacket.isUpdate()) {
-                        game.updateEnemies(enemyPacket.getEnemies());
-                    } else {
-                        game.addEnemies(enemyPacket.getEnemies());
-                    }
-                } else if (object instanceof CreatePlayerPacket && game != null) {
-                    CreatePlayerPacket packet = (CreatePlayerPacket) object;
-                    game.addPlayerShell(packet.player);
-                } else if (object instanceof DisconnectPacket && game != null) {
-                    DisconnectPacket dcPacket = (DisconnectPacket) object;
-                    game.deletePlayerShell(dcPacket.id);
-                } else if (object instanceof RoomPacket && game != null) {
-                    game.changeRoom((RoomPacket) object);
-                } else if (object instanceof ChangeEnemyStrategyPacket && game != null) {
-                    ChangeEnemyStrategyPacket enemy = (ChangeEnemyStrategyPacket) object;
-                    game.updateEnemyStrategy(enemy.id, enemy.strategy);
-                } else if (object instanceof CharacterAttackPacket && game != null) {
-                    CharacterAttackPacket character = (CharacterAttackPacket) object;
-                    game.drawPlayerAttack(character.id);
-                } else if (object instanceof ServerFullPacket) {
-                    JOptionPane.showMessageDialog(frame, "Server is full, sorry!");
-                    System.exit(0);
-                }
+                pcu.handle(client, object);
             }
         });
     }
 
-    private void handleIdPacket(IdPacket idPacket) {
-        name = idPacket.playerData.getName();
-        initUI(idPacket.currentRoom, idPacket.playerData);
-    }
-
-    private void initUI(int currentRoom, PlayerData playerData) {
+    public void initUI(int currentRoom, PlayerData playerData) {
```

Comment:
"Minimize the accessibility of classes and members"
I'd rather have the "PacketControlUnit" in the root package and have this at least as protected.

Reply:
Scratch that, looks like it has to be public since it's used in the IdPacketHandler.

Reply:
Solved.